src/utils/value-summaries.js

```diff
@@ -0,0 +1,500 @@
+    }
+  } else if (subkind === EXTERNAL_NODE_SUBKIND_ATTR) {
+    preview.value = reader.readString();
+  } else if (subkind === EXTERNAL_NODE_SUBKIND_DOCUMENT) {
+    preview.location = reader.readString();
+  } else if (subkind === EXTERNAL_NODE_SUBKIND_DOCUMENT_FRAGMENT) {
+    preview.childNodesLength = reader.readUint32();
+    if (depth < 1) {
+      preview.childNodes = [];
+      for (
+        let i = 0;
+        i < preview.childNodesLength && i < MAX_COLLECTION_VALUES;
+        i++
+      ) {
+        preview.childNodes.push(readValueSummary(reader, depth + 1, shapes));
+      }
+    }
+  } else if (
+    subkind === EXTERNAL_NODE_SUBKIND_TEXT ||
+    subkind === EXTERNAL_NODE_SUBKIND_COMMENT
+  ) {
+    preview.textContent = reader.readString();
+  }
+
+  result.preview = preview;
+}
+
+function readExternalObjectSummary(result, reader, depth, shapes) {
+  readClassFromShape(result, reader, shapes);
+
+  const startIndex = reader.getIndex();
+  const size = reader.readUint32();
+  try {
+    const version = reader.readUint8();
+    if (version !== EXTERNAL_SUMMARY_EXPECTED_VERSION) {
+      return;
+    }
+
+    const kind = reader.readUint8();
+    if (kind === EXTERNAL_SUMMARY_KIND_OTHER) {
+      return;
+    }
+
+    switch (kind) {
+      case EXTERNAL_SUMMARY_KIND_NODE: {
+        readNodeSummary(result, reader, depth, shapes);
+        break;
+      }
+      default:
+    }
+  } finally {
+    reader.setIndex(startIndex + size);
+  }
+}
+
+function readObjectSummary(reader, flags, depth, shapes) {
+  const result = {
+    type: 'object',
+    class: undefined,
+    ownPropertyLength: 0,
+    isError: false,
+    extensible: false,
+    sealed: false,
+    frozen: false,
+  };
+
+  const kind = reader.readUint8();
+  switch (kind) {
+    case OBJECT_KIND_NOT_IMPLEMENTED:
+      readClassFromShape(result, reader, shapes);
+      break;
+    case OBJECT_KIND_ARRAY_LIKE:
+      readArrayLikeSummary(result, reader, flags, depth, shapes);
+      break;
+    case OBJECT_KIND_MAP_LIKE:
+      readMapLikeSummary(result, reader, flags, depth, shapes);
+      break;
+    case OBJECT_KIND_FUNCTION:
+      readFunctionSummary(result, reader);
+      break;
+    case OBJECT_KIND_EXTERNAL:
+      readExternalObjectSummary(result, reader, depth, shapes);
+      break;
+    case OBJECT_KIND_WRAPPED_PRIMITIVE_OBJECT: {
+      result.wrappedValue = readValueSummary(reader, depth, shapes);
+      readGenericObjectSummary(result, reader, flags, depth, shapes);
+      break;
+    }
+    case OBJECT_KIND_GENERIC_OBJECT: {
+      readGenericObjectSummary(result, reader, flags, depth, shapes);
+      break;
+    }
+    case OBJECT_KIND_PROXY_OBJECT: {
+      readClassFromShape(result, reader, shapes);
+      result.preview = {
+        kind: 'Object',
+        ownProperties: Object.create(null),
+        ownPropertiesLength: 0,
+      };
+      break;
+    }
+    default:
+      throw new Error('Bad object kind');
```

Comment:
```suggestion
      throw new Error(`Bad object kind: ${kind}`);
```

(I don't know if we should keep these errors. Apart from my questions regarding them, this would make it easier to debug) (Same below)